Java/corecrypto/src/main/java/com/breadwallet/corecrypto/TransferAttribute.java

```diff
@@ -0,0 +1,82 @@
+package com.breadwallet.corecrypto;
+
+import com.breadwallet.corenative.cleaner.ReferenceCleaner;
+import com.breadwallet.corenative.crypto.BRCryptoTransferAttribute;
+import com.google.common.base.Optional;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+
+public class TransferAttribute implements com.breadwallet.crypto.TransferAttribute {
+
+    /* package */
+    static TransferAttribute create(BRCryptoTransferAttribute core) {
+        TransferAttribute attribute = new TransferAttribute(core);
+        ReferenceCleaner.register(attribute, core::give);
+        return attribute;
+    }
+
+    static TransferAttribute from(com.breadwallet.crypto.TransferAttribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+        if (attribute instanceof TransferAttribute) {
+            return (TransferAttribute) attribute;
+        }
+
+        throw new IllegalArgumentException("Unsupported TransferAmount instance");
+    }
+
+    private final BRCryptoTransferAttribute core;
+
+//    private final int value;
+    private final Supplier<String> keySupplier;
+    private final Supplier<Boolean> isRequiredSupplier;
+
+    private TransferAttribute(BRCryptoTransferAttribute core) {
+        this.core = core;
+
+        this.keySupplier = Suppliers.memoize(() -> core.getKey());
+        this.isRequiredSupplier = Suppliers.memoize(() -> core.isRequired());
+    }
+
+    @Override
+    public String getKey() {
+        return keySupplier.get();
+    }
+
+    @Override
+    public Optional<String> getValue() {
+        return Optional.of (core.getValue());
+    }
+
+    @Override
+    public void setValue(Optional<String> value) {
+        core.setValue(value.orNull());
+    }
+
+    @Override
+    public boolean isRequired() {
+        return isRequiredSupplier.get();
+    }
+
+    /* package */
+    BRCryptoTransferAttribute getCoreBRCryptoTransferAttribute() {
+        return core;
+    }
+
+    public boolean equalsOther (TransferAttribute that) {
+        return this.core == that.core || this.getKey().equals(that.getKey());
+    }
+
+    @Override
+    public boolean equals(Object that) {
+        return this == that
+                || (that instanceof TransferAttribute && equalsOther((TransferAttribute) that));
+    }
```

Comment:
IntelliJ has generators for this (command-N) that I typically use. Same with hashCode.

Reply:
My finicky style precedes IntelliJ. "-)